src/libraries/System.Net.Http/tests/FunctionalTests/HttpClientHandlerTest.Http2.cs

```diff
@@ -1763,121 +1792,135 @@ public static IEnumerable<object[]> KeepAliveTestDataSource()
         [MemberData(nameof(KeepAliveTestDataSource))]
         [ConditionalTheory(nameof(SupportsAlpn))]
         [ActiveIssue("https://github.com/dotnet/runtime/issues/41929")]
-        public async Task Http2_PingKeepAlive(TimeSpan keepAlivePingDelay, HttpKeepAlivePingPolicy keepAlivePingPolicy, bool expectRequestFail)
+        public void Http2_PingKeepAlive(TimeSpan keepAlivePingDelay, HttpKeepAlivePingPolicy keepAlivePingPolicy, bool expectRequestFail)
         {
-            TimeSpan pingTimeout = TimeSpan.FromSeconds(5);
-            // Simulate failure by delaying the pong, otherwise send it immediately.
-            TimeSpan pongDelay = expectRequestFail ? pingTimeout * 2 : TimeSpan.Zero;
-            // Pings are send only if KeepAlivePingDelay is not infinite.
-            bool expectStreamPing = keepAlivePingDelay != Timeout.InfiniteTimeSpan;
-            // Pings (regardless ongoing communication) are send only if sending is on and policy is set to always.
-            bool expectPingWithoutStream = expectStreamPing && keepAlivePingPolicy == HttpKeepAlivePingPolicy.Always;
+            RemoteExecutor.Invoke(RunTest, keepAlivePingDelay.Ticks.ToString(), keepAlivePingPolicy.ToString(), expectRequestFail.ToString()).Dispose();
 
-            TaskCompletionSource serverFinished = new TaskCompletionSource();
+            static async Task RunTest(string keepAlivePingDelayString, string keepAlivePingPolicyString, string expectRequestFailString)
+            {
+                // We should refactor this test so it can react to RTT PINGs.
+                // For now, avoid interference by disabling them:
+                AppContext.SetSwitch("System.Net.SocketsHttpHandler.Http2FlowControl.DisableDynamic2WindowSizing", true);
```

Comment:
This test has to be re-designed from scratch to deal with the occasional RTT PINGs. It was really hard for me to follow what's going on here, and it's disabled anyways, so I decided to turn off `DisableDynamic2WindowSizing` and keep the current logic.

@ManickaP @aik-jahoda if we want to do something better, I'll need some guidance/ideas.